spec/formatting.md

```diff
@@ -230,65 +231,77 @@ the following steps are taken:
 
 ### Fallback Resolution
 
-A **_fallback value_** is the resolved value emitted when an _expression_ cannot be resolved.
+A **_fallback value_** is the resolved value for an _expression_ that fails to resolve.
 
 An _expression_ fails to resolve when:
 
-- A _variable_ _operand_ fails to resolve.
+- A _variable_ used as an _operand_ (with or without an _annotation_) fails to resolve.
+  * Note that this does not include a _variable_ used as an _option_ value.
 - A _function_ _annotation_ fails to resolve.
 - A _private-use_ _annotation_ is unsupported by the implementation or if
   a _private-use_ _annotation_ fails to resolve.
 - The _expression_ has a _reserved_ _annotation_.
 
 The _fallback value_ depends on the contents of the _expression_:
 
-- _expression_ with _literal_ _operand_: U+007C VERTICAL LINE `|`
-  followed by the value of the Literal,
-  and then by U+007C VERTICAL LINE `|`.
-  The same representation is used for both _quoted_ and _unquoted_ values.
+- _expression_ with _function_ _annotation_ and no _operand_:
+  the _annotation_ starting sigil followed by its _identifier_
 
-  > Examples: `|your horse|`, `|42|`
+  > Examples:
+  > In a context where `:func` fails to resolve, `{:func}` resolves to the _fallback value_ `:func`.
+  > In a context where `:ns:func` fails to resolve, `{:ns:func}` resolves to the _fallback value_ `:ns:func`.
 
-- _expression_ with _variable_ _operand_: U+0024 DOLLAR SIGN `$`
-  followed by the _variable_ _name_ of the _operand_
+- _expression_ with non-_function_ _annotation_ and no _operand_:
+  - If the _annotation_ consists of a _sigil_-prefixed _identifier_
+    optionally followed by a whitespace-prefixed tail:
+    the _annotation_ starting sigil followed by that _identifier_
 
-  > Example: `$user`
+    > Examples:
+    > In any context, `{@reserved}` and `{@reserved |...|}` both resolve to the _fallback value_ `@reserved`.
 
-- _expression_ with no _operand_:
-  the _function_ starting sigil followed by its _name_
+  - Otherwise: the U+FFFD REPLACEMENT CHARACTER `�`
 
-  > Examples: `:platform`, `+tag`, `-tag`
+    > Examples:
+    > In any context, `{@reserved\\}` and `{? foo ?}` both resolve to the _fallback value_ `�`.
 
-- Otherwise: The U+FFFD REPLACEMENT CHARACTER `�` character
+- _expression_ with _literal_ _operand_:
+  U+007C VERTICAL LINE `|`
+  followed by the value of the _literal_
+  with escaping applied to U+005C REVERSE SOLIDUS `\` and U+007C VERTICAL LINE `|`,
+  and then by U+007C VERTICAL LINE `|`.
+  The same representation is used for both _quoted_ and _unquoted_ values.
 
-_Option_ names and values are not included in the _fallback value_.
+  > Examples:
+  > In a context where `:func` fails to resolve,
+  > `{42 :func}` resolves to the _fallback value_ `|42|` and
+  > `{|C:\\| :func}` resolves to the _fallback value_ `|C:\\|`.
```

Comment:
We preserve the escaped value (such as `\\`, `\|` or `\{`) here, but presumably these should be _unescaped_ at the end of processing (as they would be if the literal were processed normally). The `\` escape is syntax to allow the literal's actual value to contain the escaped character.

Reply:
Shouldn't this be as follows, given the text above on escaping `\`?
```suggestion
  > `{|C:\\| :func}` resolves to the _fallback value_ `|C:\\\\|`.
```

Reply:
Aah, you're right. Sorry for the noise.